test/penalizer/Utils.ts

```diff
@@ -0,0 +1,170 @@
+  verifier: TestVerifierEverythingAcceptedInstance
+  token: TestTokenInstance
+  chainId: number
+
+  constructor (
+    relayHub: RelayHubInstance,
+    relayOwner: Address,
+    relayWorker: Address,
+    relayManager: Address,
+    forwarder: IForwarderInstance,
+    verifier: TestVerifierEverythingAcceptedInstance,
+    token: TestTokenInstance,
+    chainId: number
+  ) {
+    this.relayHub = relayHub
+    this.relayOwner = relayOwner
+    this.relayWorker = relayWorker
+    this.relayManager = relayManager
+    this.forwarder = forwarder
+    this.verifier = verifier
+    this.token = token
+    this.chainId = chainId
+  }
+
+  async init (): Promise<void> {
+    await this.token.mint('1000', this.forwarder.address)
+
+    await this.relayHub.stakeForAddress(this.relayManager, 1000, {
+      from: this.relayOwner,
+      value: ether('1'),
+      gasPrice: gasPrice
+    })
+
+    await this.relayHub.addRelayWorkers([this.relayWorker], { from: this.relayManager, gasPrice: gasPrice })
+  }
+
+  async createRelayRequest (params: RelayRequestParams): Promise<RelayRequest> {
+    return {
+      request: {
+        relayHub: this.relayHub.address,
+        to: params.to,
+        data: params.relayData,
+        from: params.from,
+        nonce: (await this.forwarder.nonce()).toString(),
+        value: '0',
+        gas: '3000000',
+        tokenContract: this.token.address,
+        tokenAmount: '1',
+        tokenGas: '50000',
+        enableQos: params.enableQos ?? false
+      },
+      relayData: {
+        gasPrice: gasPrice,
+        relayWorker: this.relayWorker,
+        callForwarder: this.forwarder.address,
+        callVerifier: this.verifier.address,
+        domainSeparator: getDomainSeparatorHash(this.forwarder.address, this.chainId)
+      }
+    }
+  }
+
+  getRelayRequestSignature (relayRequest: RelayRequest, account: AccountKeypair): string {
+    const dataToSign = new TypedRequestData(
+      this.chainId,
+      this.forwarder.address,
+      relayRequest
+    )
+    return getLocalEip712Signature(
+      dataToSign,
+      account.privateKey
+    )
+  }
+
+  createReceipt (relayRequest: RelayRequest, signature: string): CommitmentReceipt {
+    const request = relayRequest.request
+    const relayData = relayRequest.relayData
+    return {
+      workerAddress: relayData.relayWorker,
+      commitment: {
+        time: 0,
+        from: request.from,
+        to: request.to,
+        data: request.data,
+        relayHubAddress: this.relayHub.address,
+        relayWorker: relayData.relayWorker,
+        enableQos: request.enableQos,
+        signature: signature
+      },
+      workerSignature: '0x00'
+    }
+  }
+
+  async signReceipt (commitmentReceipt: CommitmentReceipt): Promise<void> {
+    const c = commitmentReceipt.commitment
+    const commitment = new Commitment(c.time as Number, c.from as string, c.to as string, c.data, c.relayHubAddress as string, c.relayWorker as string, c.enableQos, c.signature)
```

Comment:
a `Commitment` object must be created here because this class contains logic that's necessary to arrive at a correct worker signature.

there's no other reason for this class being instantiated here